diff --git dist/emlib/inc/em_i2c.h dist/emlib/inc/em_i2c.h
index 453a169..5f94557 100644
--- dist/emlib/inc/em_i2c.h
+++ dist/emlib/inc/em_i2c.h
@@ -181,11 +181,12 @@ typedef enum {
   i2cTransferDone       = 0,    /**< Transfer completed successfully. */
 
   /* Transfer error codes (<0). */
-  i2cTransferNack       = -1,   /**< NACK received during transfer. */
-  i2cTransferBusErr     = -2,   /**< Bus error during transfer (misplaced START/STOP). */
-  i2cTransferArbLost    = -3,   /**< Arbitration lost during transfer. */
-  i2cTransferUsageFault = -4,   /**< Usage fault. */
-  i2cTransferSwFault    = -5    /**< SW fault. */
+  i2cTransferAddrNack   = -1,   /**< NACK received during address transfer. */
+  i2cTransferNack       = -2,   /**< NACK received during transfer. */
+  i2cTransferBusErr     = -3,   /**< Bus error during transfer (misplaced START/STOP). */
+  i2cTransferArbLost    = -4,   /**< Arbitration lost during transfer. */
+  i2cTransferUsageFault = -5,   /**< Usage fault. */
+  i2cTransferSwFault    = -6    /**< SW fault. */
 } I2C_TransferReturn_TypeDef;
 
 /*******************************************************************************
diff --git dist/emlib/src/em_i2c.c dist/emlib/src/em_i2c.c
index f659398..47fc35c 100644
--- dist/emlib/src/em_i2c.c
+++ dist/emlib/src/em_i2c.c
@@ -575,7 +575,7 @@ I2C_TransferReturn_TypeDef I2C_Transfer(I2C_TypeDef *i2c)
       case i2cStateAddrWFAckNack:
         if (pending & I2C_IF_NACK) {
           I2C_IntClear(i2c, I2C_IF_NACK);
-          transfer->result = i2cTransferNack;
+          transfer->result = i2cTransferAddrNack;
           transfer->state  = i2cStateWFStopSent;
           i2c->CMD         = I2C_CMD_STOP;
         } else if (pending & I2C_IF_ACK) {
@@ -606,7 +606,7 @@ I2C_TransferReturn_TypeDef I2C_Transfer(I2C_TypeDef *i2c)
       case i2cStateAddrWF2ndAckNack:
         if (pending & I2C_IF_NACK) {
           I2C_IntClear(i2c, I2C_IF_NACK);
-          transfer->result = i2cTransferNack;
+          transfer->result = i2cTransferAddrNack;
           transfer->state  = i2cStateWFStopSent;
           i2c->CMD         = I2C_CMD_STOP;
         } else if (pending & I2C_IF_ACK) {
