diff --git dist/platform/emlib/inc/em_gpio.h dist/platform/emlib/inc/em_gpio.h
index 203bcf38..1acdf113 100644
--- dist/platform/emlib/inc/em_gpio.h
+++ dist/platform/emlib/inc/em_gpio.h
@@ -995,10 +995,36 @@ __STATIC_INLINE void GPIO_PinLock(GPIO_Port_TypeDef port, unsigned int pin)
 GPIO_Mode_TypeDef GPIO_PinModeGet(GPIO_Port_TypeDef port,
                                   unsigned int pin);
 
-void GPIO_PinModeSet(GPIO_Port_TypeDef port,
-                     unsigned int pin,
-                     GPIO_Mode_TypeDef mode,
-                     unsigned int out);
+void GPIO_PinModeSetExt(GPIO_Port_TypeDef port,
+                        unsigned int pin,
+                        GPIO_Mode_TypeDef mode,
+                        unsigned int out,
+                        bool changeState);
+
+/***************************************************************************//**
+ * @brief
+ *   Set the mode for a GPIO pin.
+ *
+ * @param[in] port
+ *   The GPIO port to access.
+ *
+ * @param[in] pin
+ *   The pin number in the port.
+ *
+ * @param[in] mode
+ *   The desired pin mode.
+ *
+ * @param[in] out
+ *   A value to set for the pin in the DOUT register. The DOUT setting is important for
+ *   some input mode configurations to determine the pull-up/down direction.
+ ******************************************************************************/
+__STATIC_INLINE void GPIO_PinModeSet(GPIO_Port_TypeDef port,
+                                     unsigned int pin,
+                                     GPIO_Mode_TypeDef mode,
+                                     unsigned int out)
+{
+  GPIO_PinModeSetExt(port, pin, mode, out, true);
+}
 
 /***************************************************************************//**
  * @brief
diff --git dist/platform/emlib/src/em_gpio.c dist/platform/emlib/src/em_gpio.c
index 6aec2483..71573eb9 100644
--- dist/platform/emlib/src/em_gpio.c
+++ dist/platform/emlib/src/em_gpio.c
@@ -329,7 +329,8 @@ void GPIO_EM4WUExtIntConfig(GPIO_Port_TypeDef port,
 
 /***************************************************************************//**
  * @brief
- *   Set the mode for a GPIO pin.
+ *   Set the mode for a GPIO pin, with an additional option to prevent changing
+ *   the pin state.
  *
  * @param[in] port
  *   The GPIO port to access.
@@ -343,17 +344,21 @@ void GPIO_EM4WUExtIntConfig(GPIO_Port_TypeDef port,
  * @param[in] out
  *   A value to set for the pin in the DOUT register. The DOUT setting is important for
  *   some input mode configurations to determine the pull-up/down direction.
+ *
+ * @param[in] changeState
+ *   If true, pin state is set or cleared.
  ******************************************************************************/
-void GPIO_PinModeSet(GPIO_Port_TypeDef port,
-                     unsigned int pin,
-                     GPIO_Mode_TypeDef mode,
-                     unsigned int out)
+void GPIO_PinModeSetExt(GPIO_Port_TypeDef port,
+                        unsigned int pin,
+                        GPIO_Mode_TypeDef mode,
+                        unsigned int out,
+                        bool changeState)
 {
   EFM_ASSERT(GPIO_PORT_PIN_VALID(port, pin));
 
   /* If disabling a pin, do not modify DOUT to reduce the chance of */
   /* a glitch/spike (may not be sufficient precaution in all use cases). */
-  if (mode != gpioModeDisabled) {
+  if (changeState && mode != gpioModeDisabled) {
     if (out) {
       GPIO_PinOutSet(port, pin);
     } else {
@@ -371,7 +376,7 @@ void GPIO_PinModeSet(GPIO_Port_TypeDef port,
     BUS_RegMaskedWrite(&(GPIO->P[port].MODEH), 0xFu << ((pin - 8) * 4), (uint32_t)mode << ((pin - 8) * 4));
   }
 
-  if (mode == gpioModeDisabled) {
+  if (changeState && mode == gpioModeDisabled) {
     if (out) {
       GPIO_PinOutSet(port, pin);
     } else {
