diff --git dist/emlib/src/em_emu.c dist/emlib/src/em_emu.c
index 1ef42a4..5bb9952 100644
--- dist/emlib/src/em_emu.c
+++ dist/emlib/src/em_emu.c
@@ -333,7 +333,10 @@ static void __attribute__ ((noinline)) ramWFI(void)
 
 #else
   __WFI();                      // Enter EM2 or EM3
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Warray-bounds"
   *(volatile uint32_t*)4;       // Clear faulty read data after wakeup
+#pragma GCC diagnostic pop
 #endif
 }
 SL_RAMFUNC_DEFINITION_END
